client/my-sites/plugins-wpcom/jetpack-plugins-panel.jsx

```diff
@@ -0,0 +1,220 @@
+import { localize } from 'i18n-calypso';
+import {
+	identity,
+	find
+} from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import PluginIcon from 'my-sites/plugins/plugin-icon/plugin-icon';
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+import JetpackPluginItem from './jetpack-plugin-item';
+import SectionHeader from 'components/section-header';
+import SectionNav from 'components/section-nav';
+import NavTabs from 'components/section-nav/tabs';
+import NavItem from 'components/section-nav/item';
+import Search from 'components/search';
+import jetpackPlugins from './jetpack-plugins';
+
+class JetpackPluginsPanel extends Component {
+
+	static propTypes = {
+		siteSlug: PropTypes.string,
+		hasBusiness: PropTypes.bool,
+		hasPremium: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		translate: identity,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedGroup: 'all',
+			searchTerm: '',
+		};
+	}
+
+	onNavClick( group ) {
+		this.setState( {
+			selectedGroup: group,
+		} );
+	}
+
+	onNavSearch = searchTerm => {
+		this.setState( {
+			searchTerm,
+		} );
+	}
+
+	getSearchPlaceholder() {
+		const { translate } = this.props;
+		switch ( this.state.selectedGroup ) {
+			case 'all':
+				return translate( 'Search All…', { textOnly: true } );
+			case 'engagement':
+				return translate( 'Search Engagement…', { textOnly: true } );
+			case 'security':
+				return translate( 'Search Security…', { textOnly: true } );
+			case 'appearance':
+				return translate( 'Search Appearance…', { textOnly: true } );
+			case 'writing':
+				return translate( 'Search Writing…', { textOnly: true } );
+		}
+	}
+
+	getSelectedText() {
+		const found = find( this.getNavItems(), item => this.state.selectedGroup === item.key );
+		return found ? found.title : '';
+	}
+
+	getNavItems() {
+		const { translate } = this.props;
+
+		return [
+			{
+				key: 'all',
+				title: translate( 'All', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'all' ),
+			},
+			{
+				key: 'engagement',
+				title: translate( 'Engagement', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'engagement' ),
+			},
+			{
+				key: 'security',
+				title: translate( 'Security', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'security' ),
+			},
+			{
+				key: 'appearance',
+				title: translate( 'Appearance', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'appearance' ),
+			},
+			{
+				key: 'writing',
+				title: translate( 'Writing', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'writing' ),
+			},
+		];
+	}
+
+	getJetpackPlugins() {
+		const groupedPlugins = jetpackPlugins.map( group => this.filterGroup( group ) ).filter( g => g );
```

Comment:
with single parameter functions we need not specify the parameter

```js
const groupedPlugins = jetpackPlugins
	.map( this.filterGroup )
	.filter( identity );
```

also check out `lodash.groupBy` which will group items in a collection based on a function that returns the group name

```js
const pluginGroups = jetpackPlugins.groupBy( property( 'category' ) );
```

now we have an object whose keys are the category name and whose values are lists of the elements matching those categories. we can then simplify the other logic around that